src/dyn_model.jl

```diff
@@ -0,0 +1,383 @@
+    end
+
+    KKT_tot = vcat(KKT_static_tot, KKT_storage_tot)
+    @assert length(KKT_tot) == kkt_dims_dyn(n, m, l, T)
+
+    return KKT_tot
+end
+
+kkt_dyn(x, net::DynamicPowerNetwork, d) = kkt_dyn(
+    x, net.fq, net.fl, d, net.pmax, net.gmax, net.A, net.B, net.P, net.C; τ=net.τ
+)
+
+"""
+Compute the terms in the kkt matrix that are only related to the storage
+"""
+function kkt_storage(ν_next, ν_t, λsl, λsu, λdsl_next, λdsl_t, λdsu_next, λdsu_t, ds, s, P, C)
+    return [
+        (λsu - λsl) + (λdsu_t - λdsl_t) - (λdsu_next - λdsl_next) + (ν_t - ν_next);
+        λsl .* (-s);
+        λsu .* (s - C);
+        λdsu_t .* (ds - P);
+        λdsl_t .* (-ds - P);
+    ]
+end
+
+"""
+    extract_vars_t(P::PowerManagementProblem, t)
+
+Extracts both the primal and dual variables for a dynamic problem at timestep `t`.
+"""
+function extract_vars_t(P::PowerManagementProblem, t)
+
+    n, m = size(P.params.A)
+    _, l = size(P.params.B)
+    T = length(P.g)
+    
+    n_constraints_static = 5
+    n_constraints_storage = 4
+
+    g = P.g[t].value[:]
+    p = P.p[t].value[:]
+    s = P.s[t].value[:]
+
+    start_index = (t - 1) * n_constraints_static
+    λpl = P.problem.constraints[start_index + 1].dual  
+    λpu = P.problem.constraints[start_index + 2].dual 
+    λgl = P.problem.constraints[start_index + 3].dual
+    λgu = P.problem.constraints[start_index + 4].dual 
+    ν = P.problem.constraints[start_index + 5].dual  
+
+    # checking size consistency of primal variables
+    @assert length(λpl) == m
+    @assert length(λpu) == m
+    @assert length(λgl) == l
+    @assert length(λgu) == l
+    @assert length(ν) == n
+
+    storage_index = n_constraints_static * T + (t - 1) * n_constraints_storage
+    λsl = P.problem.constraints[storage_index + 1].dual 
+    λsu = P.problem.constraints[storage_index + 2].dual 
+    λdsu = P.problem.constraints[storage_index + 3].dual
+    λdsl = P.problem.constraints[storage_index + 4].dual
+
+    # checking size consistency of dual variables
+    @assert length(λdsl) == n
+    @assert length(λdsu) == n
+    @assert length(λsl) == n
+    @assert length(λsu) == n
+
+    return g, p, s, λpl, λpu, λgl, λgu, ν, λdsl, λdsu, λsl, λsu
+end
+
+"""
+    flatten_variables_dyn(P::PowerManagementProblem)
+
+Flattens the variables from a `PowerManagementProblem`, i.e.
+all variables over all timesteps are condensed into one array.
+
+Details:
+--------
+The variables are laid out in the following order: 
+- First variables from the static problem, i.e.
+[g, p, λpl, λpu, λgl, λgu, ν] at timestep t. Variables 
+from susbsequent timesteps are concatenated. 
+- Then variables from the dynamic problem, i.e.
+[s, λsl, λsu, λdsu, λdsl]. Variables from susbsequent
+timesteps are concatenated.
+"""
+function flatten_variables_dyn(P::PowerManagementProblem)
+
+    T = length(P.g)
+    n, m = size(P.params.A)
+    _, l = size(P.params.B)
+
+    x_static = Array{Float64}(undef, 0)
+    x_storage = Array{Float64}(undef, 0)
+
```

Comment:
@degleris1 I saw you changed this in the above to `Float64[]`. Is that the standard way to write it?